.github/workflows/testing.yml

```diff
@@ -5,191 +5,83 @@ on:
-    - v0.*.x
-    tags:
-    - v*
   pull_request:
   workflow_dispatch:
 
 jobs:
+
   initial-tests:
     name: ${{ matrix.name }}
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-latest
     strategy:
       fail-fast: false
-      matrix:
-        include:
-
-        - name: Python 3.10
-          os: ubuntu-latest
-          python: '3.10'
-          toxenv: py310
 
-        - name: Linters
-          os: ubuntu-latest
-          python: 3.8
-          toxenv: linters
-
-    steps:
-    - name: Checkout code
-      uses: actions/checkout@v3
-      with:
-        fetch-depth: 0
-    - name: Set up Python
-      uses: actions/setup-python@v4
-      with:
-        python-version: ${{ matrix.python }}
-    - name: Install Python dependencies
-      run: python -m pip install --progress-bar off --upgrade tox codecov
-    - name: Run tests
-      run: tox ${{ matrix.toxargs }} -e ${{ matrix.toxenv }} -- ${{ matrix.toxposargs }}
-  comprehensive-tests:
-    name: ${{ matrix.name }}
-    runs-on: ${{ matrix.os }}
-    needs: initial-tests
-    strategy:
-      fail-fast: false
       matrix:
         include:
 
-        - name: Python 3.8 with minimal dependencies
-          os: ubuntu-latest
-          python: 3.8
-          toxenv: py38-all-minimal
-
-        - name: Python 3.8, all tests, code coverage
-          os: ubuntu-latest
-          python: 3.8
-          toxenv: py38-all-cov
-
         - name: Python 3.8 (Windows)
           os: windows-latest
           python: 3.8
-          toxenv: py38-all
-          toxposargs: --durations=50
+          noxenv: tests-3.8
 
-        - name: Python 3.8 (MacOS X)
-          os: macos-latest
-          python: 3.8
-          toxenv: py38-all
-
-        - name: Python 3.9
+        - name: Python 3.9 (MacOS X)
           os: ubuntu-latest
           python: 3.9
-          toxenv: py39
+          noxenv: tests-3.9
+
+        - name: Python 3.10
+          noxenv: tests-3.10
+          noxposargs: --durations=10
+
+        - name: Linters
+          noxenv: linters
+
+        - name: Codespell
+          noxenv: codespell
+
+        - name: Import package
+          noxenv: import_package
 
     steps:
+
     - name: Checkout code
       uses: actions/checkout@v3
       with:
         fetch-depth: 0
+
     - name: Set up Python
       uses: actions/setup-python@v4
       with:
-        python-version: ${{ matrix.python }}
-    - name: Install Python dependencies
-      run: python -m pip install --progress-bar off --upgrade tox codecov
-    - name: Install language-pack-fr and tzdata
-      if: startsWith(matrix.name, 'Documentation')
-      run: sudo apt-get install graphviz pandoc
-    - name: Run tests
-      run: tox ${{ matrix.toxargs }} -e ${{ matrix.toxenv }} -- ${{ matrix.toxposargs }}
-    - name: Upload coverage to codecov
-      if: ${{ contains(matrix.toxenv,'-cov') }}
-      uses: codecov/codecov-action@v3
-      with:
-        file: ./coverage.xml
+        python-version: '3.10'
```

Comment:
This line needs reverting - as it is, it's installing 3.10 in all three currently attempted test envs. 3.8 Seems to already be there for 3.8/Windows, which is why that one isn't failing. I suspect it should fix the 3.9/MacOS failure